subsys/mgmt/mcumgr/transport/src/smp_lorawan.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2024, Jamie McCrae
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/lorawan/lorawan.h>
+#include <zephyr/mgmt/mcumgr/smp/smp.h>
+#include <zephyr/mgmt/mcumgr/transport/smp.h>
+#include <zephyr/mgmt/mcumgr/mgmt/handlers.h>
+
+#include <mgmt/mcumgr/transport/smp_internal.h>
+#include <mgmt/mcumgr/transport/smp_reassembly.h>
+
+LOG_MODULE_REGISTER(smp_lorawan, CONFIG_MCUMGR_TRANSPORT_LORAWAN_LOG_LEVEL);
+
+static void smp_lorawan_downlink(uint8_t port, bool data_pending, int16_t rssi, int8_t snr,
+				 uint8_t len, const uint8_t *hex_data);
+
+static int smp_lorawan_uplink(struct net_buf *nb);
+
+static uint16_t smp_lorawan_get_mtu(const struct net_buf *nb);
+
+static struct lorawan_downlink_cb lorawan_smp_downlink_cb = {
+	.port = CONFIG_MCUMGR_TRANSPORT_LORAWAN_FRAME_PORT,
+	.cb = smp_lorawan_downlink
```

Comment:
This would probably please the clang-format check.
```suggestion
	.cb = smp_lorawan_downlink,
```

Reply:
clang-format is completely broken in most checks which is why it does not fail CI, the suggestion from clang-format here is `+	.port = CONFIG_MCUMGR_TRANSPORT_LORAWAN_FRAME_PORT, .cb = smp_lorawan_downlink};` which looks ghastly. Later on it suggest this `+			K_KERNEL_STACK_SIZEOF(smp_lorawan_stack), smp_lorawan_uplink_thread, NULL,` which is funny because the suggestion **would** be a CI failure as that line is 102 characters